syntax = "proto3";

package spaceship;

service SpaceshipService {
  rpc GetShips (CoordinateRequest) returns (stream Spaceship);
}

message CoordinateRequest {
    double ra = 1;
    double dec = 2;
}

enum ShipClass {
  CORVETTE = 0;
  FRIGATE = 1;
  CRUISER = 2;
  DESTROYER = 3;
  CARRIER = 4;
  DREADNOUGHT = 5;
}

enum Alignment {
  ALLY = 0;
  ENEMY = 1;
}

message Officer {
  string first_name = 1;
  string last_name = 2;
  string rank = 3;
}

message Spaceship {
  string name = 1;
  Alignment alignment = 2;
  ShipClass ship_class = 3;
  int32 length = 4;
  int32 crew_size = 5;
  bool armed = 6;
  repeated Officer officers = 7;
}
